node/sync.go

```diff
@@ -124,12 +111,25 @@ EntrySyncLoop: // Syncs all eblocks we care about and their entries
 	// Apply all the effects
 	if graded != nil { // If graded was nil, then there was no oprs this eblock
 		d.Pegnet.InsertGradedBlock(graded)
+		d.Pegnet.InsertGrade(ctx, height, graded.WinnersShortHashes())
+		winners := graded.Winners()
+		if len(winners) > 0 {
+			d.Pegnet.InsertRate(ctx, height, winners[0].OPR.GetOrderedAssetsUint())
+		}
+	}
+
+	// TODO: Handle converts/txs
+
+	// update synced state
+	err = d.Pegnet.InsertSynced(ctx, height)
```

Comment:
We should use a sql tx to make sure all db writes are all or nothing per block

Reply:
yes i planned that for a future pr

Reply:
i was getting a lot of corrupted databases by just ctrl+c'ing out of the node and trying to restart it, so i ended up wrapping it in a tx

Reply:
gotcha :+1: